src/Psl/Option/Option.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Option;
+
+use Closure;
+
+/**
+ * @template T
+ */
+final class Option
+{
+    /**
+     * @param ?array{T} $option
+     *
+     * @internal
+     */
+    private function __construct(
+        private readonly null|array $option,
+    ) {
+    }
+
+    /**
+     * Create an option with some value.
+     *
+     * @template Tv
+     *
+     * @param Tv $value
+     *
+     * @return Option<Tv>
+     */
+    public static function some(mixed $value): Option
+    {
+        return new self([$value]);
+    }
+
+    /**
+     * Create an option with none value.
+     *
+     * @template Tn
```

Comment:
This is problematic, since it requires an explicit `/** @var */` everywhere

Reply:
having `@return Option<null>` wouldn't be a good idea either :/ which psalm had something like a wildcard for this specific case `@return Option<_>`

Reply:
One thing I did in my example from today was:

```php
/**
 * @template T
 * @param \Psl\Type\TypeInterface<T>
 * @return self<T>
 */
public static function empty(\Psl\Type\TypeInterface $type)
```

That can be used for type inference

Reply:
oh, that's a nice idea!

Reply:
if Option was covariant, `Option<never>` would also work: https://github.com/someniatko/php-result-type/blob/master/src/None.php

Reply:
@someniatko 
Splitting None and Some classes looks like a better approach as well here! 
That way, it's consistent with the PSL Result classes as well.

---

I wished we had built-in support for composed union types in PHP like:

```
type Option=Some<T>|None;
```

(Which is closer to None<never>)